terraform/app/variables.tf

```diff
@@ -0,0 +1,201 @@
+}
+
+variable "domain_name" {
+  type        = string
+  default     = "mavistesting.com"
+  description = "Domain for which to create DNS certificate"
+}
+
+variable "dns_certificate_arn" {
+  type        = string
+  default     = ""
+  description = "The ARN for a pre-existing DNS certificate to be used for ECS service"
+}
+
+variable "firewall_subnet_cidr" {
+  type        = string
+  description = "CIDR block for the firewall subnet"
+  default     = "10.0.5.0/24"
+}
+
+variable "enable_firewall" {
+  type        = bool
+  default     = false
+  description = "Boolean toggle to determine whether the firewall should be enabled."
+}
+
+variable "firewall_log_retention_days" {
+  type        = number
+  default     = 3
+  description = "Number of days to retain logs for the firewall"
+}
+
+
+variable "resource_name" {
+  type = object(
+    {
+      dbsubnet_group           = string
+      db_cluster               = string
+      rds_security_group       = string
+      loadbalancer             = string
+      lb_security_group        = string
+      cloudwatch_vpc_log_group = string
+    }
+  )
+  description = "Names of terraform managed resource. Used to import pre-existing infrastructure resources"
+}
+
+variable "ecs_log_retention_days" {
+  type        = number
+  default     = 7
+  description = "Number of days to retain logs for ecs instances"
+}
+variable "vpc_log_retention_days" {
+  type        = number
+  default     = 7
+  description = "Number of days to retain logs for the vpc traffic"
+}
+
+########## Task definition configuration ##########
+
+variable "rails_env" {
+  type        = string
+  default     = "development"
+  description = "The rails environment configuration to use for the mavis application"
+  validation {
+    condition     = contains(["development", "staging", "production"], var.rails_env)
+    error_message = "Incorrect rails environment, allowed values are: {development, staging, production}"
+  }
+}
+
+variable "rails_master_key_path" {
+  type        = string
+  default     = "/mavis/development/credentials/RAILS_MASTER_KEY"
+  description = "The path of the System Manager Parameter Store secure string for the rails master key."
+}
+
+variable "container_name" {
+  type        = string
+  default     = "mavis"
+  description = "Name of essential container in the task definition."
+}
+
+variable "docker_image" {
+  type        = string
+  default     = "<CHANGE_ME>"
+  description = "The docker name for the essential container in the task definition."
+}
+
+variable "image_tag" {
+  type        = string
+  description = "The docker image tag for the essential container in the task definition."
+}
+
+locals {
+  container_name = "${var.container_name}-${var.environment_string}"
+  docker_image   = var.docker_image == "<CHANGE_ME>" ? "mavis-${var.environment_string}" : var.docker_image
+  is_production  = var.environment_string == "production"
+  dev_task_envs = [
```

Comment:
Do we need to be able to deploy a `development` version of the app? I'm wondering if it would be safer to have some validation that makes sure the `environment_string` _isn't_ set to `development`?